### ISSUE TITLE
internal/modules/cjs/loader.js:638

### ISSUE THREAD

**Issue by yogendrakumar96** [nodejs/help#2347]
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module '/home/zagg/Desktop/mern-auth1/server.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
[nodemon] app crashed - waiting for file changes before starting...

**Comment by Zabrah** [comment]
Did you check, if the file permission allows reading that module?

**Comment by DavidRuby** [comment]
im running into the same issue, there is really no help out there about this topic

**Comment by Sharif111** [comment]
Hey DavidRuby try
1.Delete the node_modules directory
2.Delete the package-lock.json file
3.Run npm install
4.Run npm start
OR
rm -rf node_modules package-lock.json && npm install && npm start

**Comment by atmimran** [comment]
facing same error, anyone help?

**Comment by kallayyash** [comment]
Reinstall nodejs ---- its worked for me

**Comment by Themercee** [comment]
> Reinstall nodejs ---- its worked for me

I had version 10 and will trying to import a simple module on Windows 10 using normal local import gave me this error. Reinstalling to version 12 fix the problem!

**Comment by Aryan-Tiwari** [comment]
somebody help

**Comment by Aryan-Tiwari** [comment]
Request will not be  working if you are using version 10 for node.js or use alternatives like got instead of request.

**Comment by TimofeyZaycev** [comment]
indeed, with Node.js installation is trouble.
before, I   didn't have  such problem.
after i   remove  windows10,  I  reinstalled it.
next, I  installed  node.js
but, then, I  am getting trouble.
I help  that i am getting some missing from Node install.
I tired  node.js 12. 0 and 10.0 all
but, still it is the same.
I  want some help about this.
thanks.

**Comment by pagkly** [comment]
I have same problem

**Comment by ryzokuken** [comment]
We'd need more information to help you with this. Evidently, you're either trying to `require` a file that doesn't exist, or one that the user isn't allowed to read.

**Comment by pagkly** [comment]
> We'd need more information to help you with this. Evidently, you're either trying to `require` a file that doesn't exist, or one that the user isn't allowed to read.

I am using root account so how to solve this?

**Comment by ryzokuken** [comment]
@pagkly wait, first of all, don't use `npm` with `sudo`. Second of all, what are you trying to `require` and does the file exist?

**Comment by pagkly** [comment]
> @pagkly wait, first of all, don't use `npm` with `sudo`. Second of all, what are you trying to `require` and does the file exist?

But in other version of kali i use they work fine with root
I already reinstalled npm several times right from repo
```apt install npm```
I did not use any require at all
 calling npm -v just throw err

**Comment by ryzokuken** [comment]
so `npm -v` fails on a fresh install of `apt install nodejs`?

**Comment by pagkly** [comment]
> so `npm -v` fails on a fresh install of `apt install nodejs`?

Yes Nodejs is working but npm has that error
Yes i tried nvm as well but same result

**Comment by ryzokuken** [comment]
`/usr/bin/npm` isn't from `nvm, it's from your package manager's repos. Can you confirm that `nvm use latest && npm -v` fails with this error for you?

**Comment by pagkly** [comment]
> `/usr/bin/npm` isn't from `nvm`, it's from your package manager's repos. Can you confirm that `nvm use latest && npm -v` fails with this error for you?

if i use ```nvm install v13.12.0``` and then ```npm -v``` it shows this
![image](https://user-images.githubusercontent.com/28628331/78554093-d6d01f00-784d-11ea-8de3-721c20ac422f.png)

**Comment by ryzokuken** [comment]
Unfortunately, I cannot reproduce this on my `Linux version 5.5.13-arch2-1 (linux@archlinux) (gcc version 9.3.0 (Arch Linux 9.3.0-1))`. I'm closing this, since it doesn't seem to be a problem with node itself, but I'd recommend you report this to `nvm` instead.

**Comment by edwinb24** [comment]
>  rm -rf node_modules package-lock.json && npm install && npm start

This solved the issue for me. Thanks! @Sharif111

**Comment by AdnanRruka** [comment]
This problem is cuz u have the projct file in a driver where node is not installed.
I moved my project to from D driver to C and it works without any problem.

**Comment by ChipClark** [comment]
I uninstalled nodejs 10.14.2 - installed nodejs 12.16.3 and ran into this error. 
downloaded v10.21.0 - and still had the error. 

It wasn't until I deleted all the files in the nodejs directory and reinstalled 10.14.2 that I solved the issue.  However, I have another issue as v10.14.2 doesn't work for the application I need. 

So, I will continue to work on upgrading to a later version of nodejs

**Comment by Imprasna** [comment]
![Screenshot from 2020-12-02 15-18-38](https://user-images.githubusercontent.com/31104095/100856618-ac546580-34b1-11eb-8229-ff677b27b38d.png)
Can anyone help me fix this error

**Comment by burak4ydin** [comment]
only use the below
$ npm install
dont have to re-install

**Comment by saigorantla** [comment]
I have 5 projects that are running on the same server without any issues but facing this issue with my latest project.

**Comment by Amounaaaaa** [comment]
same issue

**Comment by FranciscoPintos** [comment]
I had the same error but reinstall doesn't work for me, my problem was simpler. The path you were trying to access had a space in the name. Ex: require("./app/models");
but the app folder for node did not exist, so I listed the names with the ls command and it showed me as 'app ', so I renamed the folder without the space and it worked.

**Comment by omprakashmankar** [comment]
The error message you've encountered is related to Node.js module resolution. It appears that the "resolve" module is missing or not installed correctly in your Node.js environment. The "resolve" module is commonly used in Node.js applications and build tools like Gulp to resolve file paths and dependencies.

To resolve this issue, you can try the following steps:

1. **Verify Node.js and npm Installation:** Ensure that Node.js and npm are installed correctly on your system. You can check their versions by running the following commands in your terminal:

   ```
   node -v
   npm -v
   ```

   If they are not installed or their versions are outdated, consider installing or updating them to the latest versions from the official Node.js website (https://nodejs.org/).

2. **Reinstall Gulp:** It's possible that there might be an issue with the Gulp installation. You can try uninstalling and reinstalling Gulp globally:

   ```
   npm uninstall -g gulp
   npm install -g gulp
   ```

3. **Check Your Project's Dependencies:** Make sure your project's `package.json` file contains the necessary dependencies, including Gulp and its plugins. You can navigate to your project's directory and run:

   ```
   npm install
   ```

   This will install the dependencies listed in your project's `package.json` file.

4. **Check for Global npm Modules:** Sometimes, global npm modules can interfere with local project modules. Check if you have any conflicting global modules and consider removing them.

5. **Node.js Environment Variables:** Ensure that your `NODE_PATH` environment variable is correctly set. It should include paths to your globally installed npm packages.

6. **Node.js Version Compatibility:** Check if the Gulp version you're using is compatible with your Node.js version. You can find compatibility information in the Gulp documentation or by checking the `engines` field in your project's `package.json` file.

After performing these steps, try running your Gulp task again. If the issue persists, please provide more details about your project setup and dependencies so that I can offer more specific assistance.

**Comment by lurchinms** [comment]
Error: Cannot find module 'node:path'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/npm/lib/cli.js:2:18)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)

**Comment by lurchinms** [comment]
even when i type npm -v

**Comment by zov-wvg** [comment]
I've got the same issue, running npm -v gives me an error. 

```
npm -v
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'node:path'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/usr/local/lib/node_modules/npm/lib/cli.js:2:18)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
```

**Comment by timerring** [comment]
`sudo apt remove nodejs` and then reinstall it that really helps me a lot.
